lib/ecto/query/builder.ex

```diff
@@ -1163,10 +1166,8 @@ defmodule Ecto.Query.Builder do
   end
 
   # Escapes an `Ecto.Query` and associated structs.
-  defp escape_query(%Query{} = query),
-    do: {:%{}, [], Map.to_list(query)}
-  defp escape_query(other),
-    do: other
+  @spec escape_query(Query.t) :: Macro.t
+  defp escape_query(%Query{} = query), do: {:%{}, [], Map.to_list(query)}
```

Comment:
Because this is the request of apply, there is no guarantee apply won't emit an already escaped AST, so we should not change this either. :)